src/lintsampler/density_structures/tree.py

```diff
@@ -506,7 +506,12 @@ def _generate_romberg_estimates(self):
             # Subtract the refined differences in place
             raws[:-1] = raws[:-1] - np.diff(raws) / divisor
             # Append the first refined estimate to the estimates list
-            estimates.append(raws[0])
+
+            # A guard for particularly bad estimates
```

Comment:
It is possible to get negative Romberg estimates. One strategy is to zero this estimate out, since we know the mass must be positive. Maybe there are other strategies or more careful handling?

Reply:
I'm surprised that `raws[0]` would every be negative: it gets computed from average density times volume, all of which should be positive!

Reply:
That's what I thought too -- I'll look into how negative `estimates` get returned.